compiler/rustc_infer/src/traits/util.rs

```diff
@@ -68,59 +68,71 @@ impl<'tcx> Extend<ty::Predicate<'tcx>> for PredicateSet<'tcx> {
 /// `T: Foo`, then we know that `T: 'static`.
 pub struct Elaborator<'tcx> {
     stack: Vec<PredicateObligation<'tcx>>,
-    visited: PredicateSet<'tcx>,
+    seen: PredicateSet<'tcx>,
 }
 
-pub fn elaborate_trait_ref<'tcx>(
-    tcx: TyCtxt<'tcx>,
-    trait_ref: ty::PolyTraitRef<'tcx>,
-) -> Elaborator<'tcx> {
-    elaborate_predicates(tcx, std::iter::once(trait_ref.without_const().to_predicate(tcx)))
+impl<'tcx> Elaborator<'tcx> {
+    // Elaborate the transitive implied predicates of an elaboratable thing `e`.
```

Comment:
```suggestion
    /// Elaborate the transitive implied predicates of an elaboratable thing `e`.
```

Could you make this comment and the next two doc comments?